homeassistant/components/zimi/coordinator.py

```diff
@@ -0,0 +1,114 @@
+"""Zimi Controller coordinator class device."""
+
+import logging
+import pprint
+
+from zcc import ControlPoint, ControlPointDescription, ControlPointError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_connect_to_coordinator(
+    host: str, port: int, fast: bool = False
+) -> ControlPoint | None:
+    """Connect to Zimi Cloud Controller with defined parameters."""
+
+    _LOGGER.debug("Connecting to %s:%d", host, port)
+
+    try:
+        api = ControlPoint(
+            description=ControlPointDescription(
+                host=host,
+                port=port,
+            )
+        )
+        await api.connect(fast=fast)
+
+    except ControlPointError as error:
+        _LOGGER.error("Connection failed: %s", error)
+        raise ConfigEntryNotReady(error) from error
+
+    if api.ready:
+        _LOGGER.debug("Connected")
+
+        if not fast:
+            api.start_watchdog()
+            _LOGGER.debug("Started watchdog")
+
+        return api
+
+    msg = "Connection failed: not ready"
+    _LOGGER.error(msg=msg)
+
+    return None
+
+
+class ZimiCoordinator(DataUpdateCoordinator):
```

Comment:
we still need to finish the discussion about the coordinator. At the moment I would argue we don't use any of the coordinator features, the device gets all its updates pushed anyways. So I would argue remove it for now and add it if you need it in a future PR.

Reply:
I see your point.   I was conflating the coordinator class with HA device, but now I can see how to create the HA device instance for the controller without a coordinator class.

Changes made and tested in https://github.com/home-assistant/core/pull/129876/commits/c7ffdd277ff7a362ab587718d6d7605de1f7d41d